great_expectations/rule_based_profiler/data_assistant_result/data_assistant_result.py

```diff
@@ -1970,53 +2014,205 @@ def _get_expect_domain_values_to_be_between_line_chart(
+            )
+            anomaly_coded_lines.append(anomaly_coded_line + anomaly_coded_points)
+
+        return alt.layer(band, lower_limit, upper_limit, *anomaly_coded_lines)
+
+    @staticmethod
+    def _get_expect_domain_values_to_be_between_bar_chart(
+        df: pd.DataFrame,
+        metric_plot_components: List[MetricPlotComponent],
+        batch_plot_component: BatchPlotComponent,
+        domain_plot_component: DomainPlotComponent,
+        expectation_kwarg_plot_components: List[ExpectationKwargPlotComponent],
+        expectation_type: Optional[str] = None,
+    ) -> alt.LayerChart:
+        expectation_kwarg_line_color: alt.HexColor = alt.HexColor(
+            ColorPalettes.HEATMAP_6.value[4]
+        )
+        expectation_kwarg_line_stroke_width: int = 5
+
+        title: alt.TitleParams = determine_plot_title(
+            expectation_type=expectation_type,
+            metric_plot_components=metric_plot_components,
             batch_plot_component=batch_plot_component,
             domain_plot_component=domain_plot_component,
         )
 
-        # encode point color based on anomalies
-        metric_name: str = metric_plot_component.name
-        predicate: Union[bool, int] = (
-            (alt.datum.min_value > alt.datum[metric_name])
-            & (alt.datum.max_value > alt.datum[metric_name])
-        ) | (
-            (alt.datum.min_value < alt.datum[metric_name])
-            & (alt.datum.max_value < alt.datum[metric_name])
-        )
-        point_color_condition: alt.condition = alt.condition(
-            predicate=predicate,
-            if_false=alt.value(Colors.GREEN.value),
-            if_true=alt.value(Colors.PINK.value),
-        )
+        expectation_kwarg_tooltips: List[alt.Tooltip] = []
+        min_value_plot_component: Optional[ExpectationKwargPlotComponent] = None
+        max_value_plot_component: Optional[ExpectationKwargPlotComponent] = None
+        for expectation_kwarg_plot_component in expectation_kwarg_plot_components:
+            expectation_kwarg_tooltips.append(
+                expectation_kwarg_plot_component.generate_tooltip()
+            )
+
+            if expectation_kwarg_plot_component.name == "min_value":
+                min_value_plot_component = expectation_kwarg_plot_component
+            elif expectation_kwarg_plot_component.name == "max_value":
+                max_value_plot_component = expectation_kwarg_plot_component
+
+        tooltip: List[alt.Tooltip] = []
+        if domain_plot_component.name is not None:
+            tooltip.append(domain_plot_component.generate_tooltip())
+
+        tooltip.extend(batch_plot_component.generate_tooltip())
+        tooltip.extend(expectation_kwarg_tooltips)
+        tooltip.extend(
+            [
+                metric_plot_component.generate_tooltip(format=",")
+                for metric_plot_component in metric_plot_components
+            ]
+        )
+
+        lower_limit: alt.Chart = (
```

Comment:
There is a lot of repitition in this file. Maybe removing some would improve readability? You could combine this like:
```
lower_limit: alt.Chart
upper_limit: alt.Chart
lower_limit, upper_limit = (
  alt.Chart(data=df)
            .mark_line(
                color=expectation_kwarg_line_color,
                strokeWidth=expectation_kwarg_line_stroke_width,
            )
            .encode(
                x=alt.X(
                    batch_plot_component.name,
                    type=batch_plot_component.alt_type,
                    title=batch_plot_component.title,
                    axis=alt.Axis(labels=False),
                ),
                y=component.plot_on_axis(),
                tooltip=tooltip,
            )
            .properties(title=title)
        )
  for component in (max_value_plot_component, min_value_plot_component))

Reply:
I agree that this is the preferred method to be more concise, although I will admit that I usually find comprehensions like this less readable 🤣 ✅